java-frontend/src/main/java/org/sonar/java/se/MethodYield.java

```diff
@@ -19,29 +19,87 @@
  */
 package org.sonar.java.se;
 
+import org.sonar.java.se.constraint.BooleanConstraint;
 import org.sonar.java.se.constraint.Constraint;
+import org.sonar.java.se.constraint.ObjectConstraint;
+import org.sonar.java.se.symbolicvalues.SymbolicValue;
 
 import javax.annotation.Nullable;
 
+import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public class MethodYield {
   Constraint[] parametersConstraints;
   int resultIndex;
   @Nullable
   Constraint resultConstraint;
+  boolean exception;
 
   public MethodYield(int arity) {
     this.parametersConstraints = new Constraint[arity];
     this.resultIndex = -1;
     this.resultConstraint = null;
+    this.exception = false;
   }
 
   @Override
   public String toString() {
     return "{params: " + Arrays.toString(parametersConstraints) + ", result: " + resultConstraint + " (" + resultIndex + ")}";
   }
 
+  public List<ProgramState> statesAfterInvocation(List<SymbolicValue> invocationArguments, ProgramState programState, Supplier<SymbolicValue> svSupplier) {
+    List<ProgramState> results = new ArrayList<>();
+    results.add(programState);
+    for (int index = 0; index < invocationArguments.size(); index++) {
+      // FIXME : varargs method should be handled
+      SymbolicValue invokedArg = invocationArguments.get(index);
+      Constraint constraint = parametersConstraints[Math.min(index, parametersConstraints.length - 1)];
+      if (constraint == null) {
+        // no constraint on this parameter, let's try next one.
+        continue;
+      }
+
+      List<ProgramState> programStates = new ArrayList<>();
+      if (constraint instanceof ObjectConstraint) {
+        for (ProgramState state : results) {
+          programStates.addAll(invokedArg.setConstraint(state, (ObjectConstraint) constraint));
+        }
+      } else if (constraint instanceof BooleanConstraint) {
+        for (ProgramState state : results) {
+          programStates.addAll(invokedArg.setConstraint(state, (BooleanConstraint) constraint));
+        }
+      }
+      if (programStates.isEmpty()) {
+        // constraint can't be satisfied, no need to process things further, this yield is not applicable.
+        // TODO there might be some issue to report in this case.
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1135)